src/types.hpp

```diff
@@ -437,6 +437,35 @@ namespace madrona_gpudrive
     const size_t ScenarioIdExportSize = 32;
     static_assert(sizeof(ScenarioId) == sizeof(char32_t) * ScenarioIdExportSize);
 
+    // Traffic light state definitions - Add to types.hpp
+    enum class TLState : int32_t
+    {
+        Unknown = 0,
+        Stop = 1,
+        Caution = 2,
+        Go = 3
+    };
+
+    struct TrafficLightState
+    {
+        // Lane ID for this traffic light
+        int32_t laneId;
+
+        // Arrays of state data for each timestep
+        TLState state[consts::kTrajectoryLength-1];
+        float x[consts::kTrajectoryLength-1];
+        float y[consts::kTrajectoryLength-1];
+        float z[consts::kTrajectoryLength-1];
+        int32_t timeIndex[consts::kTrajectoryLength-1];
+        int32_t laneIds[consts::kTrajectoryLength-1];
+
+        // Number of valid states
+        uint32_t numStates;
+    };
+
+    // 1 (TLState) + 3 (x,y,z coordinates) + 2 (timeIndex, laneId) = 6
+    const size_t TrafficLightsExportSize = 6 * consts::kTrajectoryLength-1;
```

Comment:
I think we need a `+ 2` for the `laneId` and `numStates` ?

and something like (im not sure if the `sizeof(float)` is correct, apologies if it is wrong)
```
static_assert(sizeof(TrafficLightState) == sizeof(float) * TrafficLightsExportSize);
```